src/Psalm/Internal/Analyzer/AttributesAnalyzer.php

```diff
@@ -42,6 +42,16 @@ class AttributesAnalyzer
         40 => 'promoted property',
     ];
 
+    // Copied from Attribute class since that class might not exist at runtime
+    public const TARGET_CLASS = 1;
+    public const TARGET_FUNCTION = 2;
+    public const TARGET_METHOD = 4;
+    public const TARGET_PROPERTY = 8;
+    public const TARGET_CLASS_CONSTANT = 16;
+    public const TARGET_PARAMETER = 32;
+    public const TARGET_ALL = 63;
+    public const IS_REPEATABLE = 64;
+
     /**
      * @param array<array-key, AttributeGroup> $attribute_groups
      * @param 1|2|4|8|16|32|40 $target
```

Comment:
could it be changed to an int mask now that we have constants?

Reply:
Nope, it doesn't allow arbitrary combinations. It can be `key-of<self::TARGET_DESCRIPTIONS>` though.